ReactWindows/ReactNative.Shared.Tests/Bridge/ReactBridgeTests.cs

```diff
@@ -197,6 +198,301 @@ public void ReactBridge_ValidJavaScriptResponse()
+            {
+                var executor = new MockJavaScriptExecutor
+                {
+                    OnCallFunctionReturnFlushedQueue = (module, method, args) => response
+                };
+
+                var callback = new MockReactCallback
+                {
+                    InvokeHandler = (moduleId, methodId, arguments) =>
+                    {
+                        ++called;
+                        Assert.That(arguments.Count, Is.EqualTo(1));
+                        Assert.That(arguments[0].Type, Is.EqualTo(JTokenType.Object));
+                        Assert.That(arguments[0]["foo"].Value<string>(), Is.EqualTo("hello"));
+                        Assert.That(arguments[0]["bar"].Value<double>(), Is.EqualTo(4.2));
+                        Assert.That(arguments[0]["baz"].Value<bool>(), Is.EqualTo(true));
+                    },
+                    OnBatchCompleteHandler = () => countdownEvent.Signal()
+                };
+
+                var bridge = new ReactBridge(executor, callback, nativeThread);
+
+                bridge.CallFunction("module", "method", new JArray());
+
+                await Task.Run(new Action(countdownEvent.Wait));
+                Assert.That(called, Is.EqualTo(1));
+            }
+        }
+
+        [Test]
+        public async Task ArrayReturn()
+        {
+            var jsText = "[[0],[0],[[[\"foo\", 4.2, false]]]]";
+            var response = JToken.Parse(jsText);
+            var called = 0;
+            var countdownEvent = new CountdownEvent(1);
+
+            using (var nativeThread = CreateNativeModulesThread())
+            {
+                var executor = new MockJavaScriptExecutor
+                {
+                    OnCallFunctionReturnFlushedQueue = (module, method, args) => response
+                };
+
+                var callback = new MockReactCallback
+                {
+                    InvokeHandler = (moduleId, methodId, arguments) =>
+                    {
+                        ++called;
+                        Assert.That(arguments.Count, Is.EqualTo(1));
+                        Assert.That(arguments[0].Type, Is.EqualTo(JTokenType.Array));
+                        Assert.That(arguments[0].Count(), Is.EqualTo(3));
+                        Assert.That(arguments[0][0].Value<string>, Is.EqualTo("foo"));
+                        Assert.That(arguments[0][1].Value<double>(), Is.EqualTo(4.2));
+                        Assert.That(arguments[0][2].Value<bool>(), Is.False);
+                    },
+                    OnBatchCompleteHandler = () => countdownEvent.Signal()
+                };
+
+                var bridge = new ReactBridge(executor, callback, nativeThread);
+
+                bridge.CallFunction("module", "method", new JArray());
+
+                await Task.Run(new Action(countdownEvent.Wait));
+                Assert.That(called, Is.EqualTo(1));
+            }
+        }
+
+        [Test]
+        public async Task ReturnMultipleParams()
```

Comment:
>ReturnMultipleParams [](start = 26, length = 20)

It looks like these tests may belong in a separate PR. I don't see anything specific to bridge idle/busy here.

Reply:
That being said, we should add some tests :-)

---
In reply to: [185373727](https://github.com/Microsoft/react-native-windows/pull/1726#discussion_r185373727) [](ancestors = 185373727)

Reply:
Wanted to pin the behavior first, will add appropriate assertions to the new tests in the next step.